tests/src/test/scala/com/evolutiongaming/kafka/journal/JournalPerfSpec.scala

```diff
@@ -72,46 +72,48 @@ class JournalPerfSpec extends AsyncWordSpec with JournalSuite {
 
     val key = Key.random[IO]("journal").unsafeRunSync()
 
-    lazy val append = {
+    def append(journals: Journals[IO]) = {
 
-      def append(journals: Journals[IO]) = {
+      val journal = JournalTest(journals(key), timestamp)
 
-        val journal = JournalTest(journals(key), timestamp)
-
-        val expected = {
-          val expected = for {
-            n     <- (0 to events).toList
-            seqNr <- SeqNr.min.map[Option](_ + n)
-          } yield {
-            event(seqNr)
-          }
-          Nel.fromListUnsafe(expected)
-        }
-
-        def appendNoise = {
-          (0 to events)
-            .toList
-            .parFoldMap1 { _ =>
-              val e = event(SeqNr.min)
-              for {
-                _       <- journal.append(Nel.of(e))
-                key     <- Key.random[IO]("journal")
-                journal  = JournalTest(journals(key), timestamp)
-                _       <- journal.append(Nel.of(e))
-              } yield {}
-            }
+      val expected = {
+        val expected = for {
+          n     <- (0 until events).toList
+          seqNr <- SeqNr.min.map[Option](_ + n)
+        } yield {
+          event(seqNr)
         }
+        Nel.fromListUnsafe(expected)
+      }
 
-        for {
-          _ <- journal.pointer
-          _ <- expected.groupedNel(10).foldMap { events => journal.append(events).void }
-          _ <- appendNoise
-        } yield {}
+      def appendNoise = {
+        (1 to events)
+          .toList
+          .parFoldMap1 { n =>
+            val e = event(SeqNr.unsafe(events + n))
+            for {
+              _       <- journal.append(Nel.of(e))
+              key     <- Key.random[IO]("journal")
+              journal  = JournalTest(journals(key), timestamp)
+              _       <- journal.append(Nel.of(e))
+            } yield {}
+          }
       }
 
-      journalOf(eventualJournal).use(append)
+      for {
+        _ <- journal.pointer
+        _ <- expected.groupedNel(10).foldMap { events => journal.append(events).void }
+        _ <- appendNoise
+      } yield {}
     }
 
+    val appendToJournal = for {
+      _ <- awaitResources
```

Comment:
could you please explain this a bit?

Reply:
Sure

`def append(journals: Journals[IO]) = ???` function is one that pre-fill the journal before executing tests. In previous implementation it was executed before each test (ie twice) thus adding duplicated records to the journal. This PR changes eventual-journal behaviour by failing on data-inconvenience instead of skipping such records. To avoid failures that are not part of the test, I've changed pre-fill logic execution - its run once after `IntegrationSuite.start()` done to avoid having duplicates in the journal

Reply:
maybe instead of hacking this around - we revert changes brought in hot-fix PR and just add new test case about this? otherwise this thing is getting really messy…

Reply:
I'm not hacking around here :) 

The test uses `EventualCassandra` by itself, even now after reverting temporal fix logic from `Journals` its still acts in the test. Previously the test was implemented in a way that `Journals` will deduplicate records which is not expected behaviour. To keep the test actual, I've changed pre-fill logic making it fear by executing only once.

If you see the change as over-complication I can propose adding docs (simple option) or reimplementing `JournalSuite` (probably) making pre-tests logic easily accessible from tests (it can take a while, I prefer to avoid it now)